ext/include/opentelemetry/ext/http/client/http_client_factory.h

```diff
@@ -0,0 +1,27 @@
+#pragma once
+#include "opentelemetry/ext/http/client/http_client.h"
+#ifdef WITH_CURL
+#  include "opentelemetry/ext/http/client/curl/http_client_curl.h"
+#endif
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace ext
+{
+namespace http
+{
+namespace client
+{
+class HttpClientFactory
+{
+public:
+#ifdef WITH_CURL
```

Comment:
It looks a bit odd to have `#ifdef WITH_CURL` in a generic factor interface?

Reply:
Correct, as of now, there is only one `HttpClientFactory::Create()` implementation, and we would be adding more as we add support for further clients. And the right factory method should be picked during compile time based on where it is compiled. Do you see any issues here ?